src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,37 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+
+type Props = {
+  selectedTodo: Todo | undefined;
+  viewModule: (id :number) => void
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, viewModule }) => {
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    getUser(selectedTodo?.userId)
+      .then(result => setSelectedUser(result));
```

Comment:
Let’s use async await. It gives you a simpler way to work with asynchronous promise-based code